src/AbstractTypedContainerImplementation.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\ServiceManager;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * @psalm-require-extends ServiceManager
+ */
+abstract class AbstractTypedContainerImplementation implements ContainerInterface
```

Comment:
Discussed below: `@internal` is not viable, but we should document the rough usage scope for this class